packages/dataclass-editor/src/master/DataClassMasterContent.tsx

```diff
@@ -116,18 +117,26 @@ export const DataClassMasterContent = () => {
     setSelectedField(selection);
   };
 
+  const updateDataArray = (moveIndexes: number[], toIndex: number) => {
+    const newArray = arrayMoveMultiple(dataClass.fields, moveIndexes, toIndex);
+    setDataClass({ ...dataClass, fields: [...newArray] });
+  };
+
   const updateOrder = (moveId: string, targetId: string) => {
     const selectedRows = table.getSelectedRowModel().flatRows.map(r => r.original.name);
     const moveIds = selectedRows.length > 1 ? selectedRows : [dataClass.fields[parseInt(moveId)].name];
-    const newDataClass = structuredClone(dataClass);
```

Comment:
are you sure that you can replace this by the {... } syntax?
the structureClone makes a deep copy of the object while {... } only makes a copy of the first level, everything deeper is still the "old" object.

Reply:
I think in theory it could be fine since it looks like we are only moving the fields around in these methods. However, it definitely needs to be done at least before calling `arrayMoveMultiple`.

Just in case, I would still create a `structuredClone` as early as possible and only work with the copy. A change in the future might make this necessary but then it might get forgotten.